File: framework/extracted/Loenn/entities/fountainSpirit.lua
local utils = require("utils")
local drawableSprite = require("structs.drawable_sprite")

local fountainSpirit = {}

fountainSpirit.name = "MaggyHelper/FountainSpirit"
fountainSpirit.depth = -100
fountainSpirit.fieldInformation = {
    spiritType = {
        options = {"Healing", "Platform", "Buff", "Guidance"},
        editable = false
    },
    healAmount = {
        minimumValue = 0
    },
    buffDuration = {
        minimumValue = 0
    }
}
fountainSpirit.placements = {
    {
        name = "healing",
        data = {
            spiritType = "Healing",
            healAmount = 3,
            buffDuration = 10.0
        }
    },
    {
        name = "buff",
        data = {
            spiritType = "Buff",
            healAmount = 0,
            buffDuration = 10.0
        }
    }
}

function fountainSpirit.sprite(room, entity)
    local sprite = drawableSprite.fromTexture("characters/fountain_spirit/dormant", entity)
    sprite:setPosition(entity.x, entity.y)
    sprite:setJustification(0.5, 1.0)
    sprite.color = {0.6, 0.8, 1.0, 0.8}
    return sprite
end

function fountainSpirit.selection(room, entity)
    return utils.rectangle(entity.x - 12, entity.y - 32, 24, 32)
end

return fountainSpirit

File: framework/extracted/Ahorn/entities/FountainSpirit.jl
module FountainSpirit

using ..Ahorn, Maple

@mapdef Entity "MaggyHelper/FountainSpirit" FountainSpirit(x::Integer, y::Integer, buffDuration::Number=10.0, healAmount::Integer=3, spiritType::String="Healing")

const placements = Ahorn.PlacementDict(
    "healing" => Ahorn.EntityPlacement(FountainSpirit),
    "buff" => Ahorn.EntityPlacement(FountainSpirit)
)

function Ahorn.selection(entity::FountainSpirit)
    return Ahorn.Rectangle(entity.x - 8, entity.y - 8, 16, 16)
end

function Ahorn.render(ctx::Ahorn.Cairo.CairoContext, entity::FountainSpirit, room::Maple.Room)
    Ahorn.drawRectangle(ctx, Ahorn.Rectangle(entity.x - 8, entity.y - 8, 16, 16), (0.5, 0.5, 0.5, 0.5), (0.0, 0.0, 0.0, 1.0))
end

end
